cmd/expand-pool.go

```diff
@@ -0,0 +1,624 @@
+					break
+				}
+				return
+			}
+		}
+		if serverDebugLog {
+			log.Infof("[Expand pool]: Start handling the expand pool action")
+		}
+		ticker := time.NewTicker(time.Second * 10)
+		defer ticker.Stop()
+		for {
+		loop:
+			select {
+			case <-ticker.C:
+				stats, err := loadPoolExpandStats()
+				if err != nil {
+					continue
+				}
+				if serverDebugLog {
+					log.Infof("[Expand pool]: Load expand status: %v", stats)
+				}
+				if len(stats.AfterPools) == 0 || len(stats.BeforePools) == 0 {
+					continue
+				}
+				switch stats.Status {
+				case "":
+					if stats.FileMD5 != "" || len(stats.AfterPools) != 0 || len(stats.BeforePools) != 0 {
+						// wait for new config file change
+						_ = writePoolExpandStats(&SelfPoolExpand{})
+					}
+					continue
+				case PoolExpandStatusWaitForFileChange:
+					err := stats.syncExpandPoolsStatusToPeer()
+					if err != nil {
+						continue
+					}
+					_ = stats.saveNextStatus(false)
+				case PoolExpandStatusSetEnvToRestart:
+					err := stats.syncExpandPoolsStatusToPeer()
+					if err != nil {
+						continue
+					}
+					err = stats.saveNextStatus(false)
+					if err != nil {
+						continue
+					}
+					setEnvToMinioArgs(fmt.Sprintf("%s %s",
+						strings.Join(stats.BeforePools, ","),
+						strings.Join(stats.AfterPools, ","),
+					))
+					errs := globalNotificationSys.SignalService(serviceRestart)
+					for _, err := range errs {
+						if err.Err != nil {
+							if serverDebugLog {
+								log.Error("[Expand pool]: SignalService to restart error:", err.Err)
+							}
+							goto loop
+						}
+					}
+					// restart self
+					globalServiceSignalCh <- serviceRestart
+					return
+				case PoolExpandStatusStartDecommission:
+					objectAPI := newObjectLayerFn()
+					if objectAPI == nil {
+						continue
+					}
+					z, ok := objectAPI.(*erasureServerPools)
+					if !ok {
+						continue
+					}
+					err := z.Decommission(context.Background(), 0)
+					if err != nil {
+						if serverDebugLog {
+							log.Error("[Expand pool]: Decommission error:", err)
+						}
+						continue
+					}
+					_ = stats.saveNextStatus(false)
+				case PoolExpandStatusWaitForDecommissionComplete:
+					objectAPI := newObjectLayerFn()
+					if objectAPI == nil {
+						continue
+					}
+					z, ok := objectAPI.(*erasureServerPools)
+					if !ok {
+						continue
+					}
+					dstatus, err := z.Status(GlobalContext, 0)
+					if err != nil {
+						if serverDebugLog {
+							log.Error("[Expand pool]: Decommission Status error:", err)
+						}
+						continue
+					}
+					if dstatus.Decommission == nil || !dstatus.Decommission.Complete {
+						continue
+					}
+					_ = stats.saveNextStatus(false)
+					// rename dir right now
+					fallthrough
+				case PoolExpandStatusWaitForRenameDataDir:
+					// load pre status
+					// if not renamed, rename now
+					preStats, err := loadPoolExpandStats()
+					if err != nil {
+						continue
+					}
+					if !preStats.DirHaveRenamed {
```

Comment:
Seems we can switch this... something like
```suggestion
					if preStats.DirHaveRenamed {
						continue 
					}
```